search/search_param_types.go

```diff
@@ -289,6 +296,24 @@ func (q *Query) UsesPipeline() bool {
 	return q.UsesIncludes() || q.UsesRevIncludes() || q.UsesChainedSearch() || q.UsesReverseChainedSearch()
 }
 
+// SupportsPaging returns true if the query results can be paginated, false if not.
+// In practice, most queries can be paginated, but queries for $everything or _summary cannot.
+func (q *Query) SupportsPaging() bool {
+	options := q.Options()
+
+	// not for $everything. $everything is defined as _id=<id>&_include=*&_revinclude=*
+	if options.IsIncludeAll && options.IsRevincludeAll && strings.Contains(q.Query, "_id") {
```

Comment:
I don't know how likely this is, but suppose someone did an `or` search on the `_id`:
```
_id=<id1>,<id2>,<id3>,<id4>,<id5>&_include=*&_revinclude=*
```

The code above would still flag it as an everything search that shouldn't be paged -- but if `_count` is less than the number of ids they pass in, then you need paging.